File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.py
N = int(input())
triangle = []

for _ in range(N):
    triangle.append(list(map(int, input().split())))

triangle.append([0] * (N + 1))

for i in range(N - 1, -1, -1):
    for j in range(i + 1):
        triangle[i][j] = triangle[i][j] + max(triangle[i + 1][j], triangle[i + 1][j + 1])
print(max(triangle[0]))

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] triangle) {
        for (int h = 1; h < triangle.length; h++) {
            triangle[h][0] += triangle[h - 1][0];
            for (int w = 1; w < h; w++) {
                triangle[h][w] += Math.max(triangle[h - 1][w - 1], triangle[h - 1][w]);
            }
            triangle[h][h] += triangle[h - 1][h - 1];
        }
        return Arrays.stream(triangle[triangle.length - 1]).max().getAsInt();
    }
}